index.js

```diff
@@ -0,0 +1,563 @@
+    throw new BedrockError('Update Order Failed.', errorType, details);
+  }
+};
+
+/**
+ * Gets the total cost of a PayPal order.
+ *
+ * @param {object} options - Options to use.
+ * @param {object} options.order - A PayPal order.
+ *
+ * @returns {BigNumber} The total of the PayPal order.
+ */
+const getTotalCost = ({order}) => {
+  const {purchase_units} = order;
+  // get the total for the purchase_units.
+  const total = purchase_units.reduce((accumulator, current) => accumulator.
+    plus(current.amount.value), BigNumber(0));
+  // there should only be one currency
+  const currencies = new Set(purchase_units.map(pu => pu.amount.currency_code));
+  return {total, currencies};
+};
+
+/**
+ * Compares the amount of a PayPal order and an amount.
+ *
+ * @param {object} options - Options to use.
+ * @param {object} options.order - A PayPal order.
+ * @param {BigNumber|number|string} options.expectedAmount - The
+ *   expected amount.
+ *
+ * @returns {boolean} True is the amounts equal.
+ */
+const compareAmount = ({order, expectedAmount}) => {
+  // get the total for the purchase_units.
+  const {total, currencies} = getTotalCost({order});
+  const sameAmount = total.isEqualTo(expectedAmount.value);
+  if(!sameAmount) {
+    throw new BedrockError(
+      `Unexpected amount.`,
+      Errors.Data,
+      {payPalId: order.id, expectedAmount, actualAmount: total.toString()}
+    );
+  }
+  const sameCurrency = currencies.has(expectedAmount.currency_code);
+  if(!sameCurrency) {
+    throw new BedrockError(
+      'Unexpected currency.',
+      Errors.Data,
+      {
+        public: true,
+        payPalId: order.id,
+        currencyCode: expectedAmount.currency_code
+      }
+    );
+  }
+  return true;
+};
+
+/**
+ * Checks that the existing PayPal order is there
+ * and match the expectedAmount.
+ *
+ * @param {object} options - Options to use.
+ * @param {object} options.payment - A Bedrock payment.
+ * @param {object} options.amount - A Paypal amount object.
+ * @param {BigNumber|number|string} options.expectedAmount - How is expected.
+ *
+ * @returns {object} The updated paypal order.
+ */
+const updateGatewayPaymentAmount = async (
+  {pendingPayment, updatedPayment}) => {
+  const expectedAmount = {
+    currency_code: pendingPayment.currency,
+    value: pendingPayment.amount
+  };
+  const amount = {
+    currency_code: updatedPayment.currency,
+    value: updatedPayment.amount
+  };
+  // ensure the order is still there.
+  const order = await getOrderFromPayment({payment: pendingPayment});
+  const value = formatAmount({amount});
+  compareAmount({order, expectedAmount});
+  const patch = [{
+    op: 'replace',
+    path: `/purchase_units/@reference_id=='${pendingPayment.id}'/amount`,
+    value
+  }];
+  const updatedOrder = await updateOrder({order, patch});
+  // make sure the swap occurred.
+  compareAmount({order: updatedOrder, expectedAmount: amount});
+  const payment = Object.assign(pendingPayment, updatedPayment);
```

Comment:
Better to not mutate `pendingPayment`.  Just create a new object.
```suggestion
  const payment = Object.assign({}, pendingPayment, updatedPayment);
```

Reply:
good idea and done. I did not commit your suggestion because I wanted to test it locally first before committing it to the repo. Added it here: https://github.com/digitalbazaar/bedrock-payment-paypal/pull/2/commits/42a81daf80556f5ef82c3744e34cb7533e466d79